### DISCUSSION TITLE
Action fails on private repository

### DISCUSSION THREAD

**Discussion by zapds** [retypeapp/retype#489] [Q&A]
this is my `.github/workflows/retype-action.yml` file
```
name: Publish Retype powered website to GitHub Pages
on:
  workflow_dispatch:
  push:
    branches:
      - main

jobs:
  publish:
    name: Publish to retype branch

    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2

      - uses: retypeapp/action-build@latest

      - uses: retypeapp/action-github-pages@latest
        with:
          update-branch: true
          github-token: ${{ secrets.GITHUB_TOKEN }}
```

but it fails when I commit this file to my private repository. These are the action logs:
```
Run retypeapp/action-github-pages@latest
Run ${GITHUB_ACTION_PATH}/github.sh
Retype built documentation path: /tmp/tmp.AzIhI6uQlO
Target branch: retype
Target directory: ./
Fetching remote for existing branches: done.
Creating new, orphan, 'retype' branch: cleanup, done.
Deploying files: done.
Staging deployed files for commit: done.
Checking for changes in repository: changes found.
Committing files: done.
Pushing 'retype' branch back to GitHub: error.
Error: unable to push the 'retype' branch back to GitHub.
Failed command and output:

$ git push origin HEAD

remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/.../.../': The requested URL returned error: 403
----

Aborting documentation build process.
Error: Process completed with exit code 1.
```
Can someone help?

**Comment by DannyBen** [comment]
I think `update-branch` and `github-token` do not work together.

From the [action-github-pages README](https://github.com/retypeapp/action-github-pages#github-token) 

> The action will never use the access token if (1) the branch does not exist, (2) update-branch is configured, or (3) if branch: HEAD is specified.

So - if I am reading it correctly - you need to use `branch` instead of `update-branch`, and point it to the same branch you configured for your GitHub pages. Then it should create a Pull Request when executed.


And this seems to be the relevant section: https://github.com/retypeapp/action-github-pages#push-to-a-custom-branch-and-use-the-actions-own-github-token-to-create-a-pull-request
